<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="Home page of phd student Pierre Marijon">
    <meta name="keywords" content="homepage pierre marijon research recherche assembly sequencing long-read">
    <title>Pierre Marijon</title>

    <link rel="stylesheet" href="css/pure/pure-min.css">
    <link rel="stylesheet" href="css/layouts/blog.css">
    <link rel="stylesheet" href="css/fonts/fontawesome/css/all.min.css">
    <link rel="stylesheet" href="css/fonts/academicons/css/academicons.min.css">
  </head>
  <body>

    <div id="layout" class="pure-g-r">
      <div class="sidebar pure-u">
	<div class="cover-img">
	  <div class="cover-body">
	    <header class="header">
	      <img class="avatar" alt="Pierre Marijon picture" src="img/common/pierre-avatar.jpg" width="250px"/>
	      <h1 class="brand-main">Pierre Marijon</h1>
	      <p class="tagline">PostDoc in bioinformatics</p>
	      <p>pierre.marijon [at] genomecad.fr</p>

	      <p class="">
		<ul class="nobull">
		  <li class="sidebar-text"><a href="#software">Software</a></li>
		  <li class="sidebar-text"><a href="#publications">Publications</a></li>
		  <li class="sidebar-text"><a href="#talks">Talks</a></li>
		  <li class="sidebar-text"><a href="#review">Review</li>
		  <li class="sidebar-text"><a href="#other_scientific_thing">Other Scientific thing</a></li>
		</ul>
	      </p>

	      <p class="social">
		<a href="https://github.com/natir/">
		  <span class="fa-brands fa-github fa-2xl" aria-hidden="true"></span>
		</a>
		<a href="https://genomic.social/@pierre_marijon">
		  <span class="fa-brands fa-mastodon fa-2xl" aria-hidden="true"></span>
		</a>
		<a href="https://bsky.app/profile/pierre-marijon.bsky.social">
		  <span class="fa-brands fa-bluesky fa-2xl"></span>
		</a>
		<a href="https://twitter.com/pierre_marijon">
		  <span class="fa-brands fa-twitter fa-2xl" aria-hidden="true"></span>
		</a>
		<a href="https://www.linkedin.com/in/pierre-marijon-06256b11b/">
		  <span class="fa-brands fa-linkedin fa-2xl" aria-hidden="true"></span>
		</a>
		<a href="https://orcid.org/0000-0002-6694-6873">
		  <span class="fa-brands fa-orcid fa-2xl" aria-hidden="true"></span>
		</a>
	      </p>
	    </header>
	  </div>
	</div>
      </div>

      <div class="content pure-u-1 pure-u-md-3-5">
        <div>
	  <section class="post">
	    <h2 class="post-title">Presentation</h2>
	    <p>
	      I'm currently part of the research and development team at the <a href="https://laboratoire-seqoia.fr/">SeqOIA</a> diagnostic genetic analysis laboratory. Where I'm working on the problem of interrogating large databases of variants from numerous individuals associated with clinical information.
	      I did a post-doc in Germany in <a href="https://bioinf.mpi-inf.mpg.de/homepage/index.php?&account=marschal">Tobias Marshall's</a> team, where I worked on the detection and phasing of variants using long-reads, as well as on the construction of pangenome.
	      I did my thesis in the <a href="http://cristal.univ-lille.fr/bonsai/">BONSAI</a> team, I was supervised by <a href="http://rayan.chikhi.name/">Rayan Chikhi</a> and <a href="https://www.cristal.univ-lille.fr/profil/varre">Jean Stéphane Varré</a>.
	    </p>
	    <p>
	      My current work focuses on the construction and interrogation of a database of variants containing a large number of individuals associated with clinical signs.
	      I have previously worked on long reads correction and their use for variants detection, <i>de novo</i> genomes assembly, construction and analysis of pangenomes but I'm still depply interested in these matters.
	    </p>
	  </section>

	  <section class="post">
	    <h2 class="post-title"><a name="software"></a>Software</h2>
	    <p>
	      <ul>
		<li><a href="https://github.com/natir/knot">KNOT</a>: for Knowledge Network Overlap exTraction a pipeline to investigate fragmented long-read assembly</li>
		<li><a href="https://github.com/natir/yacrd">yacrd</a>: for Yet Another Chimeric Read Detector: using all-against-all read mapping for chimeric read detection and read scrubbing</li>
		<li><a href="https://github.com/natir/fpa">fpa</a>: for Filter Pairwise Alignment: filter all-against-all read mapping, by type, length, name… fpa can convert alignment in overlap graph (<a href="https://github.com/GFA-spec/GFA-spec/blob/master/GFA1.md">GFA1</a>), index and rename reads</li>
		<li><a href="https://natir.github.io/ragroc">ragroc</a>: for Random dnA Generator and Reverse COmplement: a webpage write in webassembly to generate random dna and run reverse complement</li>
		<li><a href="https://crates.io/crates/niffler">niffler</a>: is a rust crate to provide simple and transparent support for compressed files</li>
		<li><a href="https://natir.github.io/variantplaner/">variantplaner</a>: a toolkit to quering many variants from many samples, with limited resources</li>
	      </ul>
	    </p>
	  </section>

	  <section class="post">
	    <h2 class="post-title"><a name="publications"></a>Publications</h2>
	    <p>My thesis is avaible on <a href="https://www.theses.fr/2019LILUI087">these.fr</a></p>
	    <h3>Main</h3>
	    <p>
	      <ul>
		<li><b>yacrd and fpa: upstream tools for long-read genome assembly</b> (<a href="https://doi.org/10.1093/bioinformatics/btaa262">Bioinformatics</a>, <a href="https://doi.org/10.1101/674036">Biorxiv</a>)</li>
		<li><b>Graph analysis of fragmented long-read bacterial genome assemblies</b> (<a href="https://doi.org/10.1093/bioinformatics/btz219">Bioinformatics</a>, <a href="dow/graph_analysis_of_fragmented_long-read_bacterial_genome_assemblies.pdf">Preprint</a>)</li>
	      </ul>
	    </p>

	    <h3>Contribution</h3>
	    <p>
	      <ul>
		<li><b>A draft human pangenome reference</b> (<a href="https://doi.org/10.1038/s41586-023-05896-x">Nature</a>)</li>
		<li><b>Gaps and complex structurally variant loci in phased genome assemblies</b> (<a href="https://doi.org/10.1101/gr.277334.122">Nature</a>)</li>
		<li><b>Cutevariant: a GUI-based desktop application to explore genetics variations</b> (<a href="https://doi.org/10.1093/bioadv/vbab028">Bioinformatics Advances</a>)</li>
		<li><b>Fully phased human genome assembly without parental data using single-cell strand sequencing and long reads</b> (<a href="https://doi.org/10.1038/s41587-020-0719-5">Nature Biotechnology</a>)</li>
		<li><b>A Graph-Theoretic Barcode Ordering Model for Linked-Reads</b> (<a href="https://doi.org/10.4230/LIPIcs.WABI.2020.11">WABI 2020</a>)</li>
	      </ul>
	    </p>
	  </section>

	  <section class="post">
	    <h2 class="post-title"><a name="talks"></a>Talks</h2>
	    <p>
	      <ul>
		<li><b>SAKE: What to fish in the lake ?</b> at <a href="https://bioinfo-diag.fr/blog/seminaire-2025/">BioInfoDiag 2025</a> (<a href="slides/bioinfodiag/2025.pdf">pdf</a>)</li>
		<li><b>SAKE: How to fish variants in a data lake</b> at <a href="https://assises-genetique.org/">AGHM 2024 congres</a> (<a href="slides/aghm/2024.pdf">pdf</a>)</li>
		<li><b>VariantPlaner: Quering many variant without cluster</b> at <a href="https://www.sfmpp.org/congres-sfmpp/congres-2023/">SFMPP 2023 congres</a> (<a href="slides/sfmpp/sfmpp_2023.pdf">pdf</a>)</li>
		<li><b>Correcting Long-Reads with k-mers: A Dream Comes True</b> at <a href="https://seqbim2020.sciencesconf.org/">SeqBim 2020</a> (<a href="slides/seqbim_2020/seqbim_2020.pdf">pdf</a>)</li>
		<li><b>Novel components at the periphery of long read genome assembly tools</b> slides of my PhD defense (<a href="slides/phd_defense/phd_defense.pdf">pdf</a>)</li>
		<li><b>Graph analysis of fragmented long-read bacterial genome assemblies</b>, at <a href="https://recombseq.recomb2019.org/">RECOMB-Seq 2019</a> (<a href="slides/recomb_seq_2019/recomb_seq2019.pdf">pdf</a>)</li>
		<li><b>Optimizing early steps of long-read genome assembly</b> at <a href="http://bioinfo.univ-rouen.fr/seqbio2018/">Seqbio 2018</a> (<a href="slides/seqbio_2018/seqbio2018.pdf">pdf</a>)</li>
		<li><b>Debugging long-read genome assemblies using string graph analysis</b> at <a href="http://bioinfo.cristal.univ-lille.fr/seqbio2017/">Seqbio 2017</a> (<a href="slides/seqbio_2017/seqbio2017.pdf">pdf</a>)</li>
	      </ul>
	    </p>
	  </section>

	  <section class="post">
	    <h2 class="post-title"><a name="review"></a>Review</h2>
	    <p>
	      For journal:
	      <ul>
		<li><a href="https://academic.oup.com/bioinformatics">Bioinformatics</a></li>
		<li><a href="https://bmcbioinformatics.biomedcentral.com/">BMC Bioinformatics</a></li>
		<li><a href="https://joss.theoj.org/">Journal of Open Source Software</a></li>
	      </ul>
	    </p>
	  </section>

	  <section class="post">
	    <h2 class="post-title"><a name="other_scientific_thing"></a>Other Scientific thing</h2>
	    <p>
	      <ul>
		<li>My scientific <a href="http://blog.pierre.marijon.fr/">blog</a></li>
		<li>Poster presentation at <a href="https://project.inria.fr/jobim2017">JOBIM 2017</a> <b>Debugging long-read genome assemblies using string graph analysis</b> (<a href="dow/poster_jobim2017.pdf">pdf</a>)</li>
		<li>Co-founder of <a href="http://labsquare.org/">Labsquare Community</a></li>
		<li>Author on french bioinformatics participative blog <a href="https://bioinfo-fr.net/author/pierre-marijon">bioinfo-fr</a></li>
	      </ul>
	    </p>
	  </section>
	</div>

        <div class="footer">
	  css provided by <a href="http://purecss.io">purecss.io</a>
	</div>
      </div>
    </div>
  </body>
</html>
